webapp/application.py

```diff
@@ -73,11 +76,44 @@
 cipher = Cipher(os.environ.get("APPLICATION_CRYPTO_SECRET_KEY"))
 base_url = "https://harvest.greenhouse.io/v1"
 
+directory_api_url = f'{os.environ["DIRECTORY_API_URL"]}/graphql/'
+directory_api_token = f'token {os.environ["DIRECTORY_API_TOKEN"]}'
+
 
 # Helpers
 # ===
 
 
+def _get_employee_directory_data(employee_id: str):
+    # Provide a GraphQL query
+    transport = RequestsHTTPTransport(
+        url=directory_api_url,
+        headers={"Authorization": directory_api_token},
+        use_json=True,
+        verify=False,
+    )
+
+    client = Client(transport=transport, fetch_schema_from_transport=True)
```

Comment:
Can you move the client to be initialised at the initialization of the application 

(line 81)

Reply:
Ok, so I am seeing a problem putting this outside. If no key is provided, the entire project will fail to run. `dotrun` will not work and the Github actions are going to require an API key. In the interview scheduler](https://github.com/canonical/interview-scheduler/blob/3534e41f84510201eafa4da69a45a7c1324808c2/scheduler/directory.py) we can see that it removes the depedency for poject initialization and only initializes it when the Directory API is required. In the case of careers, I don't see such a need, we only are using one endpoint, so there is no need for a class. So I think for now we can keep it inside of this function to remove the dependency?